### PR TITLE
[UI] Create index page of exam

### LINKED ISSUE #20: [UI] Index page of `exam` (`problems` tab)
[Figma](https://www.figma.com/file/EBXAMSH5BVf7xLt7bqVUKa/Exam?node-id=9%3A424)

### PR DESCRIPTION
**Pull request by edisonhello**
Preview:
image

* [x] Style of table is not finalized yet. Need to create more stub service functions to test table.
* [x] Add icon for tabs.

Resolves #20

### CODE REVIEW

**Comment by edisonhello** [comment]
Preview:
image

I changed a little about `total score` field since I don't know where is the exact position it should be.
I'll do rebase tomorrow.

**Comment by edisonhello** [comment]
Preview:
image

**Comment by edisonhello** [comment]
Issue(?):
* [ ] Navbar login status not update.
  * Logout first
  * Login, navbar not showing exam tab

File: src/app/app-routing.module.ts [hunk 1]
```diff
@@ -11,7 +15,15 @@ import {TestcasesComponent} from './testcases/testcases.component';
 const routes: Routes = [
   {path: '', component: LoginComponent},
   {path: 'problems', component: ProblemListComponent},
-  {path: 'exams', component: ExamListComponent},
+  { path: 'exams', component: ExamListComponent, },
+  {
+    path: 'exam/:examId', component: ExamIndexComponent,
```
Comment (Johnny850807) [inline src/app/app-routing.module.ts:20]: should be `path: 'exams/:examId'`
and I think `ExamHomeComponent` might be better.

File: src/app/exam-problems/exam-problems.component.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+          return ''; // use external default escaping
+        }
+      }
+    ).use(MarkdownItKatex);
+    markdownIt.renderer.rules.table_open = () => {
+      return '<table class="table">';
+    };
+    this.renderer.setProperty(element.nativeElement, 'innerHTML',
+      markdownIt.render(mdString));
+  }
```
Comment (Johnny850807) [inline src/app/exam-problems/exam-problems.component.ts:75]: I think we can't duplicate this section of code, please extract the markdown-relevant code into a `src/utils/mardownUtils.js` file 
And expose some utils function to be reused in `exam-problems.component` and `problem-description.component`

**Review by Johnny850807 — CHANGES_REQUESTED** [review]
Review done, thank you!

There are also things that you should improve:
1. make a directory `src/app/exam` and put those exam-related components under this  directory.
2. You can directly inject the `StubExamService`, so that we can test the beautiful UI in our beta page after it's merged.

**Review by Chaoyuuu — APPROVED** [review]
LGTM 👍🏻 👍🏻 👍🏻

**Comment by edisonhello** [comment]
@Johnny850807 Should I put `exam-list` into `exam` folder?

**Comment by Johnny850807** [comment]
> @Johnny850807 Should I put `exam-list` into `exam` folder?

yes

File: src/app/exam-problems/exam-problems.component.ts (hunk 2 above)
Comment (edisonhello) [reply src/app/exam-problems/exam-problems.component.ts:75]: I guess you mean `mardownUtils.ts`?

**Review by Johnny850807 — APPROVED** [review]
LGTM

**Pull request merged**